src/useFetch.ts

```diff
@@ -229,11 +229,15 @@ function useFetch<TData = any>(...args: UseFetchArgs): UseFetch<TData> {
   // onMount/onUpdate
   useEffect((): any => {
     mounted.current = true
-    if (Array.isArray(dependencies)) {
-      const methodName = requestInit.method || HTTPMethod.GET
-      const methodLower = methodName.toLowerCase() as keyof ReqMethods<TData>
-      const req = request[methodLower] as NoArgs
-      req()
+    if (
+      Array.isArray(dependencies) &&
+      (dependencies.length === 0 ||
+        dependencies.filter((d) => !isEmpty(d)).length > 0)
+    ) {
```

Comment:
@fkrafi are there any unit tests/tests relevant to this feature ? for the newly added/fixed one and existing functionality
So it will be easy to review ?